src/commands/debugCommands.ts

```diff
@@ -0,0 +1,206 @@
+      return;
+    }
+    const answer = await window.showInformationMessage(
+      `Do you want to validate the project: ${projectContext.label}? This will run the command 'dbt debug' inside this project. Do you want to continue?`,
+      PromptAnswer.YES,
+      PromptAnswer.NO,
+    );
+    if (answer === PromptAnswer.YES) {
+      try {
+        this.telemetry.sendTelemetryEvent("validateProject");
+        const project = this.dbtProjectContainer.findDBTProject(
+          projectContext.uri,
+        );
+        if (project === undefined) {
+          window.showErrorMessage(
+            `Project ${projectContext.label} was not found`,
+          );
+          return;
+        }
+        const runModelCommand = this.dbtCommandFactory.createDebugCommand(
+          project.projectRoot,
+          project.dbtProfilesDir,
+        );
+        try {
+          const runModelProcess =
+            await this.dbtProjectContainer.dbtClient?.executeCommand(
+              runModelCommand,
+            );
+          const runModelOutput = await runModelProcess.complete();
+          this.terminal.log(
+            `${runModelProcess.formatText(runModelOutput.toString())}`,
+          );
+          if (runModelOutput.includes("ERROR")) {
+            throw new Error();
+          }
+        } catch (runError) {
+          console.log(runError);
+          window.showErrorMessage(
+            "Error running dbt debug for project " +
+              projectContext.label +
+              ". Please check the output tab for more details.",
+          );
+        }
+      } catch (err) {
+        console.log(err);
+        this.telemetry.sendTelemetryError("validateProjectError", err);
+      }
+    }
+  }
+  async installDeps(projectContext: ProjectQuickPickItem | undefined) {
+    if (projectContext === undefined) {
+      // This shouldnt happen really
+      window.showErrorMessage(
+        "No project was selected, please select a project in the step 'Pick a dbt project' above.",
+      );
+      return;
+    }
+    const answer = await window.showInformationMessage(
+      `Do you want to validate the project: ${projectContext.label}? This will run the command 'dbt deps' inside this project. Do you want to continue?`,
+      PromptAnswer.YES,
+      PromptAnswer.NO,
+    );
+    if (answer === PromptAnswer.YES) {
+      try {
+        this.telemetry.sendTelemetryEvent("installDeps");
+        const project = this.dbtProjectContainer.findDBTProject(
+          projectContext.uri,
+        );
+        if (project === undefined) {
+          window.showErrorMessage(
+            `Project ${projectContext.label} was not found`,
+          );
+          return;
+        }
+
+        const depsCommand = this.dbtCommandFactory.createInstallDepsCommand(
+          project.projectRoot,
+          project.dbtProfilesDir,
+        );
+        try {
+          const dbtDepsProcess =
+            await this.dbtProjectContainer.dbtClient?.executeCommand(
```

Comment:
I think there is a public method that you can call on the project container to execute a command.

No need to make the dbtClient public

Reply:
The `addCommandToQueue` is the only public one but that hides the execution results. I need to parse the results to figure out whether it was successful or not based on prev discussion with anand (see notes).

Reply:
I can probably come up with another method that does that though and shouldnt need to make dbtclient public. i'll add that one.

Reply:
If i remember well there is `executeCommandImmediately`.